src/main/scala/com/fulcrumgenomics/bam/SortBam.scala

```diff
@@ -59,13 +59,15 @@ class SortBam
   @arg(flag='s', doc="Order into which to sort the records.") val sortOrder: SamOrder = SamOrder.Coordinate,
   @arg(flag='m', doc="Max records in RAM.") val maxRecordsInRam: Int = SamWriter.DefaultMaxRecordsInRam
 ) extends FgBioTool with LazyLogging {
+  
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
```

Comment:
Naive question, what changes when these are moved to the constructor/class body? 

Since the constructor doesn't do anything else, and nothing happens in `execute` before the assert, I would have thought that asserting at either location would be functionally the same

Reply:
Exceptions in the constructor are [`ValidationException`s](https://github.com/fulcrumgenomics/sopt/blob/fa5a8db01f6db59c9f9f2e40414677296d9ff8fc/src/main/scala/com/fulcrumgenomics/sopt/cmdline/ValidationException.scala#L30), and any exceptions (typically validation errors of the arguments) are nicely formatted on the command line, versus a huge stacktrace.  And it will print out the usage too.  But none of those things will happen when _running_ the tool